### DISCUSSION TITLE
Previewing content

### DISCUSSION THREAD

**Discussion by onordgren** [directus/directus#11301] [Archive]
Hey! Is there any clever way of having previewing of content in directus? For example, say that I have a collection of blog posts with statuses draft, staging, published, archived. Then I would like to show my blog posts on a staging and a production site. The staging site would show the ones in status staging/published and the production site only the ones in published. Where it gets a bit tricky is if I want to update a published blog post and preview the changes in the staging environment but not in production since if I put it in state staging it would be removed from production.

Anyone else had the same issue? Feels like there should be some clever way around it.

**Comment by luciandavidescu** [comment]
Don't know your specific setup and also hope I'm not missing something here (I'm assuming that you're talking about your frontend), but then that shouldn't be a problem – headless brings the benefit that you can have as many frontends as you want for the same backend, without conflict or error.
So the easiest and safest way would be to fully duplicate your frontend and then just make the minor adjustments needed for staging.

In production, you'll most likely have something like:

`if (status === 'published'){ _[show items]_ }`

Now, in the staging clone, just change it to:

`if (status === 'published' || status === 'staging') { _[show items]_ }`

**Reply by onordgren** [reply]
It's more of a problem for stuff that you already have made published but then want to change. For example, I have a blog post with status published that is visible in both staging and production frontend. Then I want to change this post and preview the changes in staging FE. If I set the status back to staging it will be removed from production FE, so that only leaves me with the option of directly saving it as published without previewing it. Or am I missing something? :)

After some digging I also found https://github.com/directus/directus/discussions/2975 which is pretty much the same thing so I guess it's a problem that might be solved in the future.

**Reply by luciandavidescu** [reply]
Ah, all right, got it now.
I saw there's this options to "save as copy" (with chosen fields only), didn't explore it yet but maybe it could be of use somehow.

**Reply by onordgren** [reply]
Yea that could definitely be on option. One alternative I thought about was to have a one to many relation on the item which contains different versions, for example:

Blog post
- versions
  - status
  - title
  - content
  - ...

Then you could create different versions of a blog post and enforce that only one of them can have the status of "published" at the same time. You could even have multiple published but with different dates for on/off time etc if you have that requirement. Downside is of course that you would need to copy/paste all fields if you want to create a new version but I guess that could be solved with some extension logic.

**Reply by benhaynes** [reply]
> So the easiest and safest way would be to fully duplicate your frontend and then just make the minor adjustments needed for staging.

While this is the safest... you can also do this within a single codebase. Basically, the codebase only shows `published` content, unless some token/auth is provided (eg: in a query string), and then the codebase allows showing `draft` content.

Also, I'd be down to discuss adding an iframe preview to Directus... it's essentially the same as above, but renders it inline. Also you can do the same thing with our "Dynamic Button" interface... but it opens in a new tab.

**Reply by onordgren** [reply]
Would the iframe help in this case though? I don't think there is anyway for the front end to fetch the unsaved data atm right? I'm basically looking for a way to handle changes to already published content and I think it's something that's on the roadmap for q3 so I'm keeping my fingers crossed 😊

**Reply by benhaynes** [reply]
Well, the iframe is good for previews if your workflow accommodates it. For "Save as Future Revision", that is a separate feature request on here (and a popular one).

**Comment by tedsteen** [comment]
Sorry for spam, but this feature request has 3 different issues and I want to make sure my comment is seen :)

The developer experience for this feature in storyblok is pretty nice.
https://www.storyblok.com/tp/add-a-headless-cms-to-svelte-in-5-minutes

Posting for inspiration

**Reply by rijkvanzanten** [reply]
hehe very much seen! Seen it thrice as a matter of fact :P 

For reference:

- https://github.com/directus/directus/discussions/11301
- https://github.com/directus/directus/discussions/5897
- https://github.com/directus/directus/discussions/11921
